### ISSUE TITLE
Add #format= to JSON::SchemaBuilder::String

### ISSUE THREAD

**Issue by warmwaffles** [parrish/json-schema_builder#1]
The spec allows one to declare 

```json
{
  "links": {
    "description": "A resource object **MAY** contain references to other resource objects (\"relationships\"). Relationships may be to-one or to-many. Relationships can be specified by including a member in a resource's links object.",
    "type": "object",
    "properties": {
      "self": {
        "description": "A `self` member, whose value is a URL for the relationship itself (a \"relationship URL\"). This URL allows the client to directly manipulate the relationship. For example, it would allow a client to remove an `author` from an `article` without deleting the people resource itself.",
        "type": "string",
        "format": "uri"
      },
      "related": {
        "$ref": "#/definitions/link"
      }
    },
    "additionalProperties": true
  }
}
```

**Comment by warmwaffles** [comment]
Had to monkey patch this in

```rb
module JSON
  module SchemaBuilder
    class String < Entity
      attribute :format
    end
  end
end
```

**Comment by warmwaffles** [comment]
This is also useful for and `id` that is a UUID

```json
{ "id": { "type": "string", "format": "uuid" } }
```

**Comment by parrish** [comment]
Released in 0.0.8.  Thanks!
